matchtree.go

```diff
@@ -965,6 +965,21 @@ func (d *indexData) newMatchTree(q query.Q) (matchTree, error) {
 			},
 		}, nil
 
+	case *query.RepoIds:
```

Comment:
I found my changes worked for me without this. I'm thinking that in `shards.go` we do the filtering and then we end up removing the `RepoIds` from the query itself before doing the query execution in `matchtree.go` so therefore this code path won't actually be exercised. I still added this because it's presumably needed for consistency and correctness if that optimization in `shards.go` does not get executed for some reason.

Reply:
Indeed in theory we don't have to do searches via shards.go. But in practice we do. So you make a good point, there may be an opportunity to simplify code in future. But for now, I would keep it working like this.